### PR TITLE
chore(core): update some files to es6 classes

### PR DESCRIPTION
**Pull request by davidchase**

### CODE REVIEW

File: packages/core/src/Queue.js [hunk 1]
```diff
@@ -4,58 +4,60 @@
 
 // Based on https://github.com/petkaantonov/deque
 
-export default function Queue (capPow2) {
-  this._capacity = capPow2 || 32
-  this._length = 0
-  this._head = 0
-}
+export default class Queue {
+  constructor(capPow2) {
```
Comment (Frikki) [inline packages/core/src/Queue.js]: I think it’s cleaner and more visible to the client to have the default value set with the parameter:

```
constructor(capPow2 = 32)
```

Comment (briancavalier) [reply packages/core/src/Queue.js]: Agreed.  @davidchase, could you change it in this PR?

File: packages/core/src/Queue.js [hunk 2]
```diff
@@ -4,58 +4,60 @@
...
+    this._capacity = capPow2 || 32
+    this._length = 0
+    this._head = 0
+  }
 
-Queue.prototype.push = function (x) {
-  var len = this._length
-  this._checkCapacity(len + 1)
+  push(x) {
+    const len = this._length;
```
Comment (briancavalier) [inline packages/core/src/Queue.js]: We've been infiltrated by semicolons 😱 (a few below, too)

Comment (davidchase) [reply packages/core/src/Queue.js]: semi-🚚 FTW 😆

**Comment by davidchase** [comment]
🍻

**Pull request merged**